### PR TITLE
Add currencies and add AsMajorUnits capability 

### PR DESCRIPTION
**Pull request by danielwitz**
fix for this [issue](https://github.com/Rhymond/go-money/issues/49)
Created this PR comparing with master instead of the one comparing with my working branch

### CODE REVIEW

**Comment by k315k1010** [comment]
@danielwitz 
Thank you for your great job❗
I'm looking forward to merging this PR 🚀

**Comment by danielwitz** [comment]
@Rhymond Its being a long time since I've opened this PR, is there a reason why this isn't merged yet? didn't get any review comments from you either

**Comment by Rhymond** [comment]
@danielwitz sorry. Thanks a lot for your PR! I will take a look at this evening.

File: currency.go [hunk 1]
```diff
@@ -20,64 +20,81 @@ var currencies = map[string]*Currency{
 	"AFN": {Decimal: ".", Thousand: ",", Code: "AFN", Fraction: 2, Grapheme: "\u060b", Template: "1 $"},
```
Comment (Rhymond) [inline currency.go]: @danielwitz I see you updated currency list. Is it official list? What is the source?

Comment (danielwitz) [reply currency.go]: @Rhymond I took it from the ruby's [money gem](https://github.com/RubyMoney/money/blob/master/config/currency_iso.json). It's well tested and commonly used.

File: formatter.go [hunk 2]
```diff
@@ -54,6 +55,11 @@ func (f *Formatter) Format(amount int64) string {
 	return sa
 }
 
+// ToMajorUnits returns float64 representing the value in sub units using the currency data
+func (f *Formatter) ToMajorUnits(amount int64) float64 {
```
Comment (Rhymond) [inline formatter.go]: One small performance thing. If `f.Fraction == 0` then just return `float64(amount)` instead of doing calculations.

Comment (k315k1010) [reply formatter.go]: @danielwitz Hi,
How do you think about this ?
Are there any matters to fix it ?

Comment (danielwitz) [reply formatter.go]: sure will fix

Comment (danielwitz) [reply formatter.go]: @Rhymond ready for another review

**Pull request merged**